processing/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -4041,7 +4041,10 @@ public Function asSingleThreaded(List<Expr> args, Expr.InputBindingInspector ins
           final Object[] rhsArray = rhsEval.castTo(lhsArrayType).asArray();
           return new ContainsConstantArray(rhsArray);
         } else {
-          final Object val = rhsEval.castTo((ExpressionType) lhsArrayType.getElementType()).value();
+          final ExpressionType lhsElementType = lhsArrayType == ExpressionType.NESTED_DATA
```

Comment:
lhsElementType seems to be nullable still, right? BTW what we'll have here if array type would be[ f.e. LONG](https://github.com/apache/druid/blob/5fcf4205e4b1f17edccc4a37cf338ed5e3047ae1/processing/src/main/java/org/apache/druid/math/expr/ExpressionType.java#L44)

Reply:
In case of Nested Data, element type is null, in other case we have the element type, right? 
So in that case, we are sending the COMPLEX type itself rather than its elementType

Reply:
Oh i see, so the last problem is UNKNOWN_COMPLEX?

To be fair i think that proper null handling would be a cleaner option here than conditional decision making